src/lib.rs

```diff
@@ -172,27 +172,27 @@ pub trait PolynomialCommitment<F: Field>: Sized {
     fn open<'a>(
         ck: &Self::CommitterKey,
         labeled_polynomials: impl IntoIterator<Item = &'a LabeledPolynomial<F>>,
-        commitments: impl IntoIterator<Item = &'a LabeledCommitment<Self::Commitment>>,
+        commitments: impl IntoIterator<Item = &'a LabeledCommitment<Self::Commitment>> + Clone,
         point: F,
         rands: impl IntoIterator<Item = &'a LabeledRandomness<Self::Randomness>>,
         rng: Option<&mut dyn RngCore>,
-        fs_rng: &mut Self::RandomOracle,
     ) -> Result<Self::Proof, Self::Error>
         where
             Self::Randomness: 'a,
             Self::Commitment: 'a,
     {
-        // TODO: we need to setup our own Fiat-Shamir rng, and initialize it (or absorb)
-        // using the 1) commitments  2) query point, before passing the rng to the low-
-        // level open_individual_... function.
+        let mut fs_rng = Self::RandomOracle::from_seed(&to_bytes![
+            point,
+            commitments.clone().into_iter().collect::<Vec<_>>()
+        ].unwrap());
```

Comment:
If we keep for now assuming that the low-level functions already have the full statment of the opening proof bound to the state of the fs_rng, then we need to absorb the evaluations here, too.

Reply:
Previous functions state is restored as Daniele requested bellow